compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs

```diff
@@ -135,6 +135,14 @@ featureChoiceObservers = Feature
     , featureCppFlag = "DAML_CHOICE_OBSERVERS"
     }
 
+featureTypeInterning :: Feature
+featureTypeInterning = Feature
+    { featureName = "Type interning"
+    -- TODO Change as part of #7139
+    , featureMinVersion = versionDev
+    , featureCppFlag = "DAML_TYPE_INTERNING"
```

Comment:
Does it make sense to give this feature a CPP flag? Maybe we should make the CPP flag field optional (in a separate PR).

Reply:
It makes absolutely no sense to give this feature a CPP flag. I agree, let's clean this up in a follow-up PR. There should be more features that don't need CPP flags as well.